### PR TITLE
Fixing bugs in integration

### PR DESCRIPTION
**Pull request by pashakondratyev**

### CODE REVIEW

File: packages/legacy-client/src/client/objectSets/OsdkObjectSet.test.ts [hunk 1]
```diff
@@ -243,12 +245,12 @@ describe("OsdkObjectSet", () => {
         pageToken: "fakePageToken",
       }),
     );
-    expect(result.type).toEqual("ok");
+    const selectedObjects = unwrapResultOrThrow(result);
```
Comment (mfedderly) [inline packages/legacy-client/src/client/objectSets/OsdkObjectSet.test.ts:248]: Its kind of sad that `selectedObjects` is unused and also doesn't have a `_` prefix

File: packages/legacy-client/src/client/objects/convertWireToOsdkObject.ts [hunk 2]
```diff
@@ -68,7 +69,7 @@ function createPrototype<
 
   // add the relevant keys for the link types associated with this object type
   for (
-    const [k, { multiplicity, targetType }] of Object.entries(objDef.links)
+    const [k, { multiplicity }] of Object.entries(objDef.links)
```
Comment (mfedderly) [inline packages/legacy-client/src/client/objects/convertWireToOsdkObject.ts:72]: Can we stop serializing `targetType` entirely into the OntologyDefinition? I guess we still use it for type inference?

File: packages/generator/src/util/test/createMockMinimalFiles.ts [hunk 3]
```diff
@@ -32,7 +32,6 @@ export function createMockMinimalFiles() {
     dumpFilesToConsole: () => {
       for (const [path, contents] of Object.entries(getFiles())) {
         console.group(path);
-        console.log(contents);
```
[the remark this answers is not in the document]
Comment (pashakondratyev) [reply packages/generator/src/util/test/createMockMinimalFiles.ts:35]: Oh yeah, we can probably put that back in now

File: packages/legacy-client/src/client/objects/convertWireToOsdkObject.ts (hunk 2 above)
Comment (pashakondratyev) [reply packages/legacy-client/src/client/objects/convertWireToOsdkObject.ts:72]: Yeah I think we'd still need it for type inference, and also just readability of the definition

File: packages/legacy-client/src/client/objectSets/OsdkObjectSet.test.ts (hunk 1 above)
Comment (pashakondratyev) [reply packages/legacy-client/src/client/objectSets/OsdkObjectSet.test.ts:248]: Was using that to be able to see what the error was quickly, but probably should have a eslint rule for no-unused-vars

**Review by mfedderly — APPROVED** [review]

**Pull request merged**